common/types.go

```diff
@@ -67,6 +67,19 @@ type ReadParams struct {
 	Deleted bool // optional, defaults to false
 }
 
+// WriteParams defines how we are writing data from a SaaS API.
+type WriteParams struct {
+	// The name of the object we are writing, e.g. "Account"
+	ObjectName string // required
+
+	// The external id of the object instance we are updating
+	ObjectId string // optional
```

Comment:
Can you make it more clear that if ObjectId is present then it's an update, otherwise is a create?